Sources/IO/Geometry/GLTFImporter/index.js

```diff
@@ -224,34 +263,65 @@ function extractCellBufferDataUint32(
   return indicesCellArray;
 }
 
-function calculatePrimitiveCellSize(mode) {
-  switch (mode) {
-    case MeshPrimitiveModes.POINTS:
-      return 1;
-    case MeshPrimitiveModes.LINES:
-    case MeshPrimitiveModes.LINE_LOOP:
-    case MeshPrimitiveModes.LINE_STRIP:
-      return 2;
-    case MeshPrimitiveModes.TRIANGLES:
-    case MeshPrimitiveModes.TRIANGLE_STRIP:
-    case MeshPrimitiveModes.TRIANGLE_FAN:
-      return 3;
-    default:
-      throw new Error('Invalid primitive.mode value, must be between 0 and 6');
+function extractDataArrayFloat(
+  inputBuffer,
+  byteOffset,
+  byteStride,
+  count,
+  numberOfComponents
+) {
+  const accessorBegin = byteOffset;
+  const accessorEnd = byteOffset + count * byteStride;
+  const size = 4;
+
+  const values = [];
+  for (let iter = accessorBegin; iter !== accessorEnd; iter += byteStride) {
+    let val;
+
+    for (
+      let elemIter = iter;
+      elemIter !== iter + numberOfComponents * size;
+      elemIter += 4
+    ) {
+      val = inputBuffer.getFloat32(elemIter, true);
+      values.push(val);
+    }
```

Comment:
Have you considered using `TypedArray.set()` (you might want to try `https://jsbench.me/` to check performances though).

You could get rid of the 2 nested loops if numberOfComponents is 4